### DISCUSSION TITLE
how to determine a folder size - disk space consumption (including snapshots)

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#562] [Q&A]
Hi
I have a folder structure like that
```
parent
- work
- snapshots
- - snapshot 1 of work
- - snapshot 2 of work
- .....
```
`ls` reports on `parent` 4102147, which I believe would translate as 1021TB
on `work` 4002772 which is correct - about 27TB
on `snapshots` 4099374 - 993TB
and then there are 36 snapshots - each one reporting roughly 27TB

it all would make sense it these weren't snapshots, but they are
`dirinfo` is not of much help, it effectively shows the same number (or multiplied by 3x if that's real size)

how could I get to gauge the actual space consumed?

**Comment by xandrus** [comment]
Hi,

I believe that you are looking for `mfsdirinfo -p` option. Check the information from the man page:
```
To count stats more precisely you can use -p switch which turns on slow, recursive procedure.
In this mode nodes and chunks are counted only once and the system uses current number of chunk
copies instead of goal/labels. System administrators might consider using mfsmetadirinfo (see mfsmetadirinfo(8)).
```
